force_wfmanager/left_side_pane/variable_names_registry.py

```diff
@@ -19,22 +20,28 @@ class VariableNamesRegistry(HasStrictTraits):
     #: layer added. NOT all the variables.
     #: For example, a situation like::
     #:
-    #:     [["foo", "bar"], [], ["hello"]]
+    #:     [[{'name': "pikachu", 'type': "electric"},
+    #:       {'name': "squirtle", 'type': "water"],
+    #:     [],
+    #      [{'name': "scyther", 'type': "bug"}]]
     #:
     #: means:
-    #: - the first layer has available "foo" and "bar".
-    #: - the second layer has available "foo" and "bar", because the first
-    #: layer added nothing (hence the empty second list)
-    #: - the third layer has available "foo", "bar" and "hello".
+    #: - the first layer has available "pikachu" and "squirtle".
+    #: - the second layer has available "pikachu" and "squirtle",
+    #: because the first layer added nothing (hence the empty second list)
+    #: - the third layer has available "pikachu", "squirtle" and "scyther".
     #:
     #: The size of the base list should be the number of layers plus one.
     #: the last one being the variables that are added by the last layer.
-    available_variables_stack = List(List(Identifier))
+
+    available_variables_stack = List(List(Dict(Identifier, CUBAType)))
```

Comment:
this is getting too big to parse. Try to extract a trait like LayerVariables. It might help with understanding.